extensions/amp-next-page/0.2/service.js

```diff
@@ -227,17 +220,24 @@ export class NextPageService {
   }
 
   /**
-   * @return {!PositionObserver}
-   * @private
+   * Makes sure that all pages preceding the current page are
+   * marked hidden if they are out of the viewport
+   * @param {number} index index of the page to start at
    */
-  getPositionObserver_() {
-    // For testing
-    if (this.injectedPositionObserver_) {
-      return this.injectedPositionObserver_;
+  hidePreviousPages(index) {
+    let i = index + this.scrollDirection_;
+    while (i >= 0 && i < this.pages_.length) {
+      const page = this.pages_[i];
+      const shouldBeHidden =
+        page.relativePos === ViewportRelativePos.LEAVING_VIEWPORT ||
+        page.relativePos === ViewportRelativePos.OUTSIDE_VIEWPORT ||
+        page === this.initialPage_;
+      if (shouldBeHidden && page.isVisible()) {
+        page.setVisibility(VisibilityState.HIDDEN);
+      }
+      // Go back one more page in the opposite direction of the scroll
```

Comment:
This is a bit confusing to me. If scroll direction is positive, does that mean moving up the page and thus you want to hide later pages in the array? Could you make this a bit more explicit? It seems like there is a double negation going on here, which is confusing.